### ISSUE TITLE
Failed to get workspaces: [Error: ENOENT: no such file or directory, scandir '']

### ISSUE THREAD

**Issue by merewords** [thomas-pedersen/cursor-chat-browser#4]
Hi

Running on Windows11 Version 10.0.22621 Build 22621  I get this error despite the workspaces existing in the autodetected and configured workspaceStorage location

The link just displays
![image](https://github.com/user-attachments/assets/cca99488-21f7-4fc0-a3c0-7a2f297f650c)

Here is the configuration page
![image](https://github.com/user-attachments/assets/4f0783d7-17c1-4120-9959-f7d5c157a27a)

File explorer confirming the workspaces exits in this location
![image](https://github.com/user-attachments/assets/71826f35-a55d-478a-8d28-0dc11daac64f)

And the next-server output
![image](https://github.com/user-attachments/assets/c04de29a-d57a-4380-9815-bb804974b653)

Any idea what causes this error?

Marc

**Comment by duythvn** [comment]
following as  I got the same error

**Comment by johanndiedrick** [comment]
+1 I get the same error! On MacOS

**Comment by merewords** [comment]
Thank you @zixinzuofo for the pointer...
I worked around this issue as follows:

**Install** dotenv (if not already installed):
`npm install dotenv`

**Create** a .env File (if it doesn’t exist) in the root of this project folder and add your workspaceStorage path - this is mine:
`WORKSPACE_PATH=C:\Users\MW\AppData\Roaming\Cursor\User\workspaceStorage`

Load the .env File in your route.ts: **Add** this line at the top of the file to load environment variables:
`import 'dotenv/config';`

Access the Variable: The route.ts will now read the value of WORKSPACE_PATH from .env using the original code:
(nothing to change if you have the following near the top of route.ts)
`const workspacePath = process.env.WORKSPACE_PATH || '';`

This got it working for me until someone finds a better fix

Marc

**Comment by runekaagaard** [comment]
Yeah this works on linux, thx guys!:

```bash
WORKSPACE_PATH=~/.config/Cursor/User/workspaceStorage npm run dev
```

**Comment by jmpavlick** [comment]
I'm on macOS.

I noticed that I needed to _both_ set `WORKSPACE_DIRECTORY` to my fully-qualified path - i.e., given that my local user is `john`, I needed to set it to `/Users/john/Library/Application Support/Cursor/User/workspaceStorage` -

and I needed to set the `Cursor Workspace Path` value in the webapp config to the same value.

Using a `~/` to reference my home folder seems to not work; I suspect that it's not being expanded. Or something. I don't know.

But from `zsh`, I run

```
WORKSPACE_DIRECTORY="/Users/john/Library/Application Support/Cursor/User/workspaceStorage" npm run dev
```

and it works!

**Comment by thomas-pedersen** [comment]
a fix has been merged, let me know if still an issue

**Comment by evnb** [comment]
I still experienced this issue on Windows 11 but it worked after running `$WORKSPACE_PATH="/mnt/c/Users/USERNAME/AppData/Roaming/Cursor/User/workspaceStorage"` in my powershell terminal. Thanks @zixinzuofo for the fix and @thomas-pedersen for making this :)

**Comment by seapagan** [comment]
> a fix has been merged, let me know if still an issue

Yeah i still get this issue under Ubuntu 24.04, though the fix below works:

```console
WORKSPACE_PATH=~/.config/Cursor/User/workspaceStorage npm run dev
```

**Comment by freemedom** [comment]
for windows, you can do this to add environment variable.

![Image](https://github.com/user-attachments/assets/db76a0d3-b483-4ad6-851a-3c51c5ac4b8f)
